executor/src/witgen/generator.rs

```diff
@@ -131,7 +134,8 @@ where
                         }
                     };
 
-                    progress |= self.handle_eval_result(result, strategy);
+                    progress |=
+                        self.handle_eval_result(result, strategy, || format!("{}", identity));
```

Comment:
I think this has a very hard performance impact. We don't want to convert every identity to a string just because it fails.

Reply:
Ah sorry, it's a closure.